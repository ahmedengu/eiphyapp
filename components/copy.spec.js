import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import Copy from './copy';

function setup() {
  return mount(<Copy link="http://localhost:2020" />);
}

describe('Copy Tests', () => {
  it('should work', () => {
    const actual = setup();
    expect(actual).toBe(actual);
  });

  it('should match snapshot', () => {
    const actual = setup();
    expect(actual.html()).toMatchSnapshot();
  });

  it('click on button', () => {
    document.execCommand = jest.fn();
    const actual = setup();

    act(() => {
      actual.find('button').simulate('click');
    });

    expect(document.execCommand).toBeCalledTimes(1);
    expect(document.execCommand).lastCalledWith('copy');
  });
});
