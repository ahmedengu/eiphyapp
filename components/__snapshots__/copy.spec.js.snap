// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy Tests should match snapshot 1`] = `
<div class="img-viewer__copy">
  <input class="form-control"
         type="text"
         dir="ltr"
         value="http://localhost:2020"
  >
  <button class="btn btn-outline-dark"
          type="button"
  >
    Copy
  </button>
</div>
`;
